### PR TITLE
Update nimservice status when deployment fails

### PR DESCRIPTION
**Pull request by visheshtanksale**

### CODE REVIEW

**Review by shivamerla — COMMENTED** [review]
lgtm, can we add some tests with different statuses and conditions.

**Review by slu2011 — DISMISSED** [review]

File: internal/controller/platform/standalone/nimservice_test.go [hunk 1]
```diff
@@ -406,3 +407,166 @@ var _ = Describe("NIMServiceReconciler for a standalone platform", func() {
 		})
 	})
 })
+
+var _ = Describe("isDeploymentReady for setting status on NIMService", func() {
```
Comment (shivamerla) [inline internal/controller/platform/standalone/nimservice_test.go:411]: Can we add these as a new context under existing Describe spec?  That way we don't have to initialize all these again. The Context can be "When creating the NIMService deployment" and Tests can be "It should set the status/conditions correctly when progress deadline exceeded" etc.

**Comment by visheshtanksale** [comment]
> lgtm, can we add some tests with different statuses and conditions.

@shivamerla Added unit tests

File: internal/controller/platform/standalone/nimservice_test.go (hunk 1 above)
Comment (visheshtanksale) [reply internal/controller/platform/standalone/nimservice_test.go]: Good point. Fixed it.

**Review by shivamerla — APPROVED** [review]
LGTM! Thanks @visheshtanksale

**Pull request merged**